src/libraries/icubmod/embObjMotionControl/eomcParser.cpp

```diff
@@ -1256,7 +1318,10 @@ bool Parser::getCorrectPidForEachJoint(PidInfo *ppids/*, PidInfo *vpids*/, TrqPi
     for(int i=0; i<_njoints; i++)
     {
         if(tpids[i].enabled)
+        {
             firstjoint = i;
+            break;
+        }
```

Comment:
Introduced a break on the `firstjoint` loop, so now it correctly gets the first joint (instead of the last enabled one)